### PR TITLE
Add HAProxy integration

### PR DESCRIPTION
**Pull request by kvch**
image

image

image

image

Logs dashboard

image

### CODE REVIEW

**Comment by andresrc** [comment]
What's missing here? Please remember to mark the package as experimental

File: dev/import-beats-resources/haproxy/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+The `log` dataset collects the HAProxy application logs.
+
+{{fields "log"}}
+
+## Metrics
+
+### info
+
+The HAProxy `info` metricset collects general information about HAProxy processes.
```
Comment (mtojek) [inline dev/import-beats-resources/haproxy/docs/README.md:23]: dataset

File: packages/haproxy/dataset/log/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    type: text
+    title: Paths
+    multi: true
+    required: true
+    show_user: true
+    default:
+    - /var/log/haproxy.log
+  template_path: log.yml.hbs
+  title: 'HAProxy application logs'
+  description: 'Collect HAProxy application logs using log input'
```
Comment (mtojek) [inline packages/haproxy/dataset/log/manifest.yml:17]: maybe: `Collect HAProxy application logs from files`

File: packages/haproxy/dataset/log/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    default: localhost
+  - name: syslog_port
+    type: integer
+    title: 'Syslog Port'
+    multi: false
+    required: true
+    show_user: true
+    default: 9001
+  template_path: syslog.yml.hbs
+  title: 'HAProxy application logs over syslog'
```
Comment (mtojek) [inline packages/haproxy/dataset/log/manifest.yml:35]: Did you verify if it works in both modes?

File: dev/import-beats-resources/haproxy/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+An example event for `info` looks as following:
+
+```$json
```
Comment (mtojek) [inline dev/import-beats-resources/haproxy/docs/README.md:27]: Actually example events can be stored in external files, see: https://github.com/elastic/integrations/blob/master/packages/aws/dataset/billing/sample_event.json

While rendering target docs, these files will be picked up (`{{event "billing"}}`)

File: packages/haproxy/manifest.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      show_user: true
+      default:
+      - tcp://127.0.0.1:14567
+    - name: username
+      type: text
+      title: Username
+      multi: false
+      required: false
+      show_user: false
+      default: admin
```
Comment (mtojek) [inline packages/haproxy/manifest.yml:51]: Question about default values: are username and password consistent with HAProxy default settings? I mean, if user runs a default installation of HAProxy. Are these values empty or set to "admin/admin"?

File: packages/haproxy/dataset/log/manifest.yml (hunk 3 above)
Comment (kvch) [reply packages/haproxy/dataset/log/manifest.yml:35]: Yes, I did. I did not provide a screenshot, as I did not think it added value to the PR.

File: packages/haproxy/manifest.yml (hunk 5 above)
Comment (kvch) [reply packages/haproxy/manifest.yml:51]: By default, HAProxy does not enable any stat endpoints. When users add it, they can opt for a stat endpoint without any authentication. Thus, the "username" and "password" are not required. The username and password are usually set to admin in various HAProxy guides. So it is more like a convention, not something prepackaged. Also, this is how Metricbeat sets the values for the HAProxy metricset.

Comment (mtojek) [reply packages/haproxy/manifest.yml:51]: SGTM! Thanks for the explanation.

**Pull request merged**